### ISSUE TITLE
No way to handle connection refused. (or any tcp errors) in init

### ISSUE THREAD

**Issue by ConnorRigby** [suvash/hulaaki#21]
If the Mqtt broker either goes down, or if the mqtt broker is not running upon applicaion initialization, a bad pattern match error apears causeing the client and all relying modules to crash. 
Right now i trap the exit and just back off until the connection is up again. 

Being able to get the status on Client.connect(pid, options) would be an easy fix for this. 
it would return :ok if the connection works, or {:error, reason} if not?
relevant stack trace

```
00:00:10.859 [error] GenServer #PID<0.290.0> terminating
** (stop) exited in: GenServer.call(#PID<0.292.0>, {:connect, %Hulaaki.Message.Connect{clean_session: 1, client_id: "5d2e9cca-aa43-40d3-b8b9-736cba481f06", keep_alive: 500, password: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJhZG1pbkBhZG1pbi5jb20iLCJpYXQiOjE0NzQ2NTM5NjgsImp0aSI6ImQ0OGE2OTVjLTdjNDctNGZhYS05MTRhLTY2MGQzZGFiNWY2ZSIsImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6MzAwMCIsImV4cCI6MTQ3NDk5OTU2OCwibXF0dCI6IjE5Mi4xNjguMjkuMTU0IiwiYm90IjoiNWQyZTljY2EtYWE0My00MGQzLWI4YjktNzM2Y2JhNDgxZjA2In0.FwbA4Or7HKQ-YdtPAShyWKHKsbr_DmWLjdZeZKq_-Z4UiP8CXCWzXfnI2SmigjxgOVPB4eP1R6oj4suUDpxrjAxd3S7E3wxFsEFr1dDm-DgAXO2CkFYRbvy8fD-9jlVSo-AfJmCW6rixbCnHp8-Yholgy_OLvnyIv7lPsm85SE4pbWi3BmFT72mUU8XMmB_fK_qXlO_GId8NMP1YS8vVsQNjMN91P0RLnTMo9eWHEp3oZ6zo65uCFRSpLL4DEMMwBe_qm4Gp-XpF5jsRC3hF0y-vin8QzZt6-_sU8n6JtxJb7QPAdMKL8eTO38NZIqGedPisWtcAXVKFRya0-RV5FQ", type: :CONNECT, username: "5d2e9cca-aa43-40d3-b8b9-736cba481f06", will_message: "", will_qos: 0, will_retain: 0, will_topic: ""}, [host: "192.168.29.154", port: 1883, timeout: 5000]}, 5000)
    ** (EXIT) an exception was raised:
        ** (MatchError) no match of right hand side value: {:error, :econnrefused}
            (hulaaki) lib/hulaaki/connection.ex:147: Hulaaki.Connection.open_tcp_socket/1
            (hulaaki) lib/hulaaki/connection.ex:97: Hulaaki.Connection.handle_call/3
            (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
            (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
            (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
    (elixir) lib/gen_server.ex:604: GenServer.call/3
    (fw) lib/hulaaki/client.ex:79: Mqtt.Client.handle_call/3
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:connect, [client_id: "5d2e9cca-aa43-40d3-b8b9-736cba481f06", username: "5d2e9cca-aa43-40d3-b8b9-736cba481f06", password: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJhZG1pbkBhZG1pbi5jb20iLCJpYXQiOjE0NzQ2NTM5NjgsImp0aSI6ImQ0OGE2OTVjLTdjNDctNGZhYS05MTRhLTY2MGQzZGFiNWY2ZSIsImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6MzAwMCIsImV4cCI6MTQ3NDk5OTU2OCwibXF0dCI6IjE5Mi4xNjguMjkuMTU0IiwiYm90IjoiNWQyZTljY2EtYWE0My00MGQzLWI4YjktNzM2Y2JhNDgxZjA2In0.FwbA4Or7HKQ-YdtPAShyWKHKsbr_DmWLjdZeZKq_-Z4UiP8CXCWzXfnI2SmigjxgOVPB4eP1R6oj4suUDpxrjAxd3S7E3wxFsEFr1dDm-DgAXO2CkFYRbvy8fD-9jlVSo-AfJmCW6rixbCnHp8-Yholgy_OLvnyIv7lPsm85SE4pbWi3BmFT72mUU8XMmB_fK_qXlO_GId8NMP1YS8vVsQNjMN91P0RLnTMo9eWHEp3oZ6zo65uCFRSpLL4DEMMwBe_qm4Gp-XpF5jsRC3hF0y-vin8QzZt6-_sU8n6JtxJb7QPAdMKL8eTO38NZIqGedPisWtcAXVKFRya0-RV5FQ", host: "192.168.29.154", port: 1883, timeout: 5000, keep_alive: 500], #PID<0.292.0>}
State: %{parent: #PID<0.289.0>}

00:00:10.862 [info]  Application fw exited: Fw.start(:normal, []) returned an error: shutdown: failed to start child: Controller
    ** (EXIT) shutdown: failed to start child: MqttSupervisor
        ** (EXIT) shutdown: failed to start child: MqttHandler
            ** (EXIT) an exception was raised:
                ** (MatchError) no match of right hand side value: {:error, :econnrefused}
                    (hulaaki) lib/hulaaki/connection.ex:147: Hulaaki.Connection.open_tcp_socket/1
                    (hulaaki) lib/hulaaki/connection.ex:97: Hulaaki.Connection.handle_call/3
                    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
                    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
                    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3

```

**Comment by suvash** [comment]
Hi @ConnorRigby ! Thanks for opening this PR. I'm curious if you're using this in production.

**Comment by ConnorRigby** [comment]
Yeah i am currently use it. It's not quite production yet but could be soon. 
[here](https://github.com/FarmBot-Labs/laughing-octo-telegram)

**Comment by suvash** [comment]
@ConnorRigby Just saw the link you posted. Really cool to see the library put into use. :beers:
I might have some time this summer to work on it a bit more. Let me know what shortcomings you've run into, that could be helpful.

**Comment by suvash** [comment]
Fixed in #23
